WARNING:tensorflow:From nmt/utils/iterator_utils.py:235: group_by_window (from tensorflow.contrib.data.python.ops.grouping) is deprecated and will be removed in a future version.
Instructions for updating:
Use `tf.data.experimental.group_by_window(...)`.
WARNING:tensorflow:From nmt/model_helper.py:402: __init__ (from tensorflow.python.ops.rnn_cell_impl) is deprecated and will be removed in a future version.
Instructions for updating:
This class is deprecated, please use tf.nn.rnn_cell.LSTMCell, which supports all the feature this cell currently has. Please replace the existing code with tf.nn.rnn_cell.LSTMCell(name='basic_lstm_cell').
2019-11-29 07:47:16.141779: I tensorflow/core/kernels/lookup_util.cc:376] Table trying to initialize from file /home/bcc/thcf16/tmp/iwslt15/vocab.en is already initialized.
2019-11-29 07:47:16.141810: I tensorflow/core/kernels/lookup_util.cc:376] Table trying to initialize from file /home/bcc/thcf16/tmp/iwslt15/vocab.de is already initialized.
2019-11-29 07:47:16.141857: I tensorflow/core/kernels/lookup_util.cc:376] Table trying to initialize from file /home/bcc/thcf16/tmp/iwslt15/vocab.en is already initialized.
2019-11-29 07:55:32.476761: I tensorflow/core/kernels/lookup_util.cc:376] Table trying to initialize from file /home/bcc/thcf16/tmp/iwslt15/vocab.de is already initialized.
2019-11-29 07:55:32.476814: I tensorflow/core/kernels/lookup_util.cc:376] Table trying to initialize from file /home/bcc/thcf16/tmp/iwslt15/vocab.en is already initialized.
2019-11-29 07:55:32.476839: I tensorflow/core/kernels/lookup_util.cc:376] Table trying to initialize from file /home/bcc/thcf16/tmp/iwslt15/vocab.en is already initialized.
2019-11-29 07:55:32.607591: I tensorflow/core/kernels/lookup_util.cc:376] Table trying to initialize from file /home/bcc/thcf16/tmp/iwslt15/vocab.de is already initialized.
2019-11-29 07:55:32.607641: I tensorflow/core/kernels/lookup_util.cc:376] Table trying to initialize from file /home/bcc/thcf16/tmp/iwslt15/vocab.en is already initialized.
2019-11-29 07:55:32.607641: I tensorflow/core/kernels/lookup_util.cc:376] Table trying to initialize from file /home/bcc/thcf16/tmp/iwslt15/vocab.en is already initialized.
2019-11-29 07:55:35.879587: W tensorflow/core/framework/op_kernel.cc:1273] OP_REQUIRES failed at matmul_op.cc:478 : Resource exhausted: OOM when allocating tensor with shape[8064,50000] and type float on /job:localhost/replica:0/task:0/device:CPU:0 by allocator cpu
Traceback (most recent call last):
  File "/usr/lib/python2.7/runpy.py", line 174, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "/usr/lib/python2.7/runpy.py", line 72, in _run_code
    exec code in run_globals
  File "/nobackup/bcc/thcf16/nmt/nmt/nmt.py", line 707, in <module>
    tf.app.run(main=main, argv=[sys.argv[0]] + unparsed)
  File "/nobackup/bcc/thcf16/nmt/venv/local/lib/python2.7/site-packages/tensorflow/python/platform/app.py", line 125, in run
    _sys.exit(main(argv))
  File "/nobackup/bcc/thcf16/nmt/nmt/nmt.py", line 700, in main
    run_main(FLAGS, default_hparams, train_fn, inference_fn)
  File "/nobackup/bcc/thcf16/nmt/nmt/nmt.py", line 693, in run_main
    train_fn(hparams, target_session=target_session)
  File "nmt/train.py", line 577, in train
    eval_model, eval_sess, model_dir, hparams, summary_writer)
  File "nmt/train.py", line 102, in run_internal_eval
    summary_writer, "dev")
  File "nmt/train.py", line 665, in _internal_eval
    ppl = model_helper.compute_perplexity(model, sess, label)
  File "nmt/model_helper.py", line 654, in compute_perplexity
    output_tuple = model.eval(sess)
  File "nmt/model.py", line 346, in eval
    return sess.run(output_tuple)
  File "/nobackup/bcc/thcf16/nmt/venv/local/lib/python2.7/site-packages/tensorflow/python/client/session.py", line 929, in run
    run_metadata_ptr)
  File "/nobackup/bcc/thcf16/nmt/venv/local/lib/python2.7/site-packages/tensorflow/python/client/session.py", line 1152, in _run
    feed_dict_tensor, options, run_metadata)
  File "/nobackup/bcc/thcf16/nmt/venv/local/lib/python2.7/site-packages/tensorflow/python/client/session.py", line 1328, in _do_run
    run_metadata)
  File "/nobackup/bcc/thcf16/nmt/venv/local/lib/python2.7/site-packages/tensorflow/python/client/session.py", line 1348, in _do_call
    raise type(e)(node_def, op, message)
tensorflow.python.framework.errors_impl.ResourceExhaustedError: OOM when allocating tensor with shape[8064,50000] and type float on /job:localhost/replica:0/task:0/device:CPU:0 by allocator cpu
	 [[node dynamic_seq2seq/decoder/output_projection/Tensordot/MatMul (defined at nmt/model.py:526)  = MatMul[T=DT_FLOAT, transpose_a=false, transpose_b=false, _device="/job:localhost/replica:0/task:0/device:CPU:0"](dynamic_seq2seq/decoder/output_projection/Tensordot/Reshape, dynamic_seq2seq/decoder/output_projection/kernel/read)]]
Hint: If you want to see a list of allocated tensors when OOM happens, add report_tensor_allocations_upon_oom to RunOptions for current allocation info.


Caused by op u'dynamic_seq2seq/decoder/output_projection/Tensordot/MatMul', defined at:
  File "/usr/lib/python2.7/runpy.py", line 174, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "/usr/lib/python2.7/runpy.py", line 72, in _run_code
    exec code in run_globals
  File "/nobackup/bcc/thcf16/nmt/nmt/nmt.py", line 707, in <module>
    tf.app.run(main=main, argv=[sys.argv[0]] + unparsed)
  File "/nobackup/bcc/thcf16/nmt/venv/local/lib/python2.7/site-packages/tensorflow/python/platform/app.py", line 125, in run
    _sys.exit(main(argv))
  File "/nobackup/bcc/thcf16/nmt/nmt/nmt.py", line 700, in main
    run_main(FLAGS, default_hparams, train_fn, inference_fn)
  File "/nobackup/bcc/thcf16/nmt/nmt/nmt.py", line 693, in run_main
    train_fn(hparams, target_session=target_session)
  File "nmt/train.py", line 466, in train
    eval_model = model_helper.create_eval_model(model_creator, hparams, scope)
  File "nmt/model_helper.py", line 180, in create_eval_model
    extra_args=extra_args)
  File "nmt/attention_model.py", line 64, in __init__
    extra_args=extra_args)
  File "nmt/model.py", line 95, in __init__
    res = self.build_graph(hparams, scope=scope)
  File "nmt/model.py", line 393, in build_graph
    self._build_decoder(self.encoder_outputs, encoder_state, hparams))
  File "nmt/model.py", line 526, in _build_decoder
    logits = self.output_layer(outputs.rnn_output)
  File "/nobackup/bcc/thcf16/nmt/venv/local/lib/python2.7/site-packages/tensorflow/python/layers/base.py", line 374, in __call__
    outputs = super(Layer, self).__call__(inputs, *args, **kwargs)
  File "/nobackup/bcc/thcf16/nmt/venv/local/lib/python2.7/site-packages/tensorflow/python/keras/engine/base_layer.py", line 757, in __call__
    outputs = self.call(inputs, *args, **kwargs)
  File "/nobackup/bcc/thcf16/nmt/venv/local/lib/python2.7/site-packages/tensorflow/python/keras/layers/core.py", line 963, in call
    outputs = standard_ops.tensordot(inputs, self.kernel, [[rank - 1], [0]])
  File "/nobackup/bcc/thcf16/nmt/venv/local/lib/python2.7/site-packages/tensorflow/python/ops/math_ops.py", line 2985, in tensordot
    ab_matmul = matmul(a_reshape, b_reshape)
  File "/nobackup/bcc/thcf16/nmt/venv/local/lib/python2.7/site-packages/tensorflow/python/ops/math_ops.py", line 2057, in matmul
    a, b, transpose_a=transpose_a, transpose_b=transpose_b, name=name)
  File "/nobackup/bcc/thcf16/nmt/venv/local/lib/python2.7/site-packages/tensorflow/python/ops/gen_math_ops.py", line 4560, in mat_mul
    name=name)
  File "/nobackup/bcc/thcf16/nmt/venv/local/lib/python2.7/site-packages/tensorflow/python/framework/op_def_library.py", line 787, in _apply_op_helper
    op_def=op_def)
  File "/nobackup/bcc/thcf16/nmt/venv/local/lib/python2.7/site-packages/tensorflow/python/util/deprecation.py", line 488, in new_func
    return func(*args, **kwargs)
  File "/nobackup/bcc/thcf16/nmt/venv/local/lib/python2.7/site-packages/tensorflow/python/framework/ops.py", line 3274, in create_op
    op_def=op_def)
  File "/nobackup/bcc/thcf16/nmt/venv/local/lib/python2.7/site-packages/tensorflow/python/framework/ops.py", line 1770, in __init__
    self._traceback = tf_stack.extract_stack()

ResourceExhaustedError (see above for traceback): OOM when allocating tensor with shape[8064,50000] and type float on /job:localhost/replica:0/task:0/device:CPU:0 by allocator cpu
	 [[node dynamic_seq2seq/decoder/output_projection/Tensordot/MatMul (defined at nmt/model.py:526)  = MatMul[T=DT_FLOAT, transpose_a=false, transpose_b=false, _device="/job:localhost/replica:0/task:0/device:CPU:0"](dynamic_seq2seq/decoder/output_projection/Tensordot/Reshape, dynamic_seq2seq/decoder/output_projection/kernel/read)]]
Hint: If you want to see a list of allocated tensors when OOM happens, add report_tensor_allocations_upon_oom to RunOptions for current allocation info.


ters, time 0.09s
# Start step 0, lr 1, Fri Nov 29 07:47:16 2019
# Init train iterator, skipping 0 elements
  step 100 lr 1 step-time 0.52s wps 5.78K ppl 47777.66 gN 64.09 bleu 0.00, Fri Nov 29 07:48:08 2019
  step 200 lr 1 step-time 0.47s wps 6.53K ppl 2643.26 gN 15.36 bleu 0.00, Fri Nov 29 07:48:55 2019
  step 300 lr 1 step-time 0.49s wps 6.29K ppl 1577.35 gN 9.27 bleu 0.00, Fri Nov 29 07:49:44 2019
  step 400 lr 1 step-time 0.50s wps 6.17K ppl 1302.19 gN 9.11 bleu 0.00, Fri Nov 29 07:50:34 2019
  step 500 lr 1 step-time 0.49s wps 6.16K ppl 1096.91 gN 7.98 bleu 0.00, Fri Nov 29 07:51:23 2019
  step 600 lr 1 step-time 0.50s wps 6.16K ppl 1000.51 gN 8.06 bleu 0.00, Fri Nov 29 07:52:13 2019
  step 700 lr 1 step-time 0.50s wps 6.15K ppl 840.39 gN 6.42 bleu 0.00, Fri Nov 29 07:53:02 2019
  step 800 lr 1 step-time 0.50s wps 6.17K ppl 774.89 gN 5.56 bleu 0.00, Fri Nov 29 07:53:52 2019
  step 900 lr 1 step-time 0.49s wps 6.12K ppl 725.03 gN 5.87 bleu 0.00, Fri Nov 29 07:54:41 2019
  step 1000 lr 1 step-time 0.50s wps 6.15K ppl 674.41 gN 6.08 bleu 0.00, Fri Nov 29 07:55:32 2019
# Save eval, global step 1000
  loaded infer model parameters from tmp/nmt_attention_model_DE_s20000_l2_u64/translate.ckpt-1000, time 0.11s
  # 723
    src: Eine Architekturgeschichte mit Höhen , Tiefen und Happy End : Der österreichische Weltausstellungspavillon von Brüssel 1958 errang Preise , wurde in Wien zum Museum , stand leer und verkam .
    ref: An architectural history with highs , lows and a happy ##AT##-##AT## ever ##AT##-##AT## after . Austria &apos;s Pavilion for the 1958 World &apos;s Fair in Brussels won prizes , became a museum in Vienna , stood empty and was abandoned .
    nmt: <unk> is a <unk> .
  loaded eval model parameters from tmp/nmt_attention_model_DE_s20000_l2_u64/translate.ckpt-1000, time 0.10s
